nixos/modules/config/users-groups.nix

```diff
@@ -139,6 +139,12 @@ let
         description = "The user's home directory.";
       };
 
+      homeMode = mkOption {
+        type = types.int;
+        default = 448;
+        description = "The user's home directory mode in numeric format. See chmod(1). Note that chmod's standard octets syntax has to be converted to base 10.  For example the octal number 700 has to be specified as 448.";
```

Comment:
Just a suggestion. But for convenience, wouldnt it be nice to be able to also specify the mode in octal for easy debugging? Perhaps another option could be made which takes the mode as a string and autofills this one. (again, just a suggestion)

Reply:
:+1: , a string with some validation might cause less confusion than using decimal, which is unusual for file permissions.

`security.wrappers` pretty much uses "anything chmod can take" - either alphabetical description, or octal (as a string).

https://github.com/NixOS/nixpkgs/blob/715dc137b08213aabbbe0965b78ab938e5d8d3b7/nixos/modules/security/wrappers/default.nix#L12-L20

Reply:
I agree that octal is more natural. I opted for `Int` because it's already used for example for`services.transmission.settings.umask` and because the symbolic notation will lead to a non-reproducible result. For example specifying `g+r` doesn't guarantee that the group (and other) write flag is not set.

Should we opt for numeric octal represented as a string?

Reply:
I believe leaving it as an octal string is good enough for now (once it suits your needs). Also, as you said, we should set the mode rather than just modifying it.

Perhaps ,eventually, it would be useful to have the ability to specify `g+r`. But  we could add that ability later on since it would just expand the set of values which the option could take.